pkg/apis/serving/v1beta1/revision_defaults.go

```diff
@@ -39,7 +39,7 @@ func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
 	cfg := config.FromContextOrDefaults(ctx)
 
 	// Default TimeoutSeconds based on our configmap
-	if rs.TimeoutSeconds == nil {
+	if rs.TimeoutSeconds == nil || *rs.TimeoutSeconds == 0 {
 		ts := cfg.Defaults.RevisionTimeoutSeconds
```

Comment:
since you're here you can also change rs.TimeoutSeconds = ptr.Int64(cfg.Default.RevisionTimeoutSeconds)

Reply:
Updated/ Thank you @vagababov